src/main/java/org/apache/mesos/hdfs/state/LiveState.java

```diff
@@ -34,8 +36,18 @@ public boolean isNameNode2Initialized() {
     return !nameNode2TaskMap.isEmpty() && nameNode2TaskMap.values().iterator().next();
   }
 
-  public void addStagingTask(Protos.TaskID taskId) {
+  public HashMap<String, String> getJournalNodeNames() { return journalNodeNames; }
+
+  public HashMap<String, String> getNameNodeNames() { return nameNodeNames; }
+
+  public void addStagingTask(Protos.TaskID taskId, String taskName) {
     stagingTasks.add(taskId);
+    if (taskId.getValue().contains(HDFSConstants.NAME_NODE_TASKID)) {
+      nameNodeNames.put(taskId.getValue(), taskName);
+    }
+    if (taskId.getValue().contains(HDFSConstants.JOURNAL_NODE_ID)) {
```

Comment:
Nits: else if? Any reason to check the NNs before the JNs? So far, we've been maintaining a JN,NN,DN ordering to match the launch order.